tests/phpunit/tests/fonts/font-library/wpFontsDir.php

```diff
@@ -69,4 +69,39 @@ function set_new_values( $defaults ) {
 
 		$this->assertSame( static::$dir_defaults, $font_dir, 'The wp_get_font_dir() method should return the default values.' );
 	}
+
+	public function test_fonts_dir_filters_do_not_trigger_infinite_loop() {
+		/*
+		 * Naive filtering of uploads directory to return font directory.
+		 *
+		 * This emulates the approach a plugin developer may take to
+		 * add the filter when extending the font library functionality.
+		 */
+		add_filter( 'upload_dir', 'wp_get_font_dir' );
+
+		add_filter(
+			'upload_dir',
+			function ( $upload_dir ) {
```

Comment:
```suggestion
			static function ( $upload_dir ) {
```

Reply:
The function uses the `$this` object so can't be a static function. This the commit and revert.